runtime/gc_base/GCExtensions.cpp

```diff
@@ -311,29 +311,36 @@ MM_GCExtensions::releaseNativesForContinuationObject(MM_EnvironmentBase* env, j9
 }
 
 bool
-MM_GCExtensions::needScanStacksForContinuationObject(J9VMThread *vmThread, j9object_t objectPtr, bool isGlobalGC)
+MM_GCExtensions::needScanStacksForContinuationObject(J9VMThread *vmThread, j9object_t objectPtr, bool isConcurrentGC, bool isGlobalGC, bool beingMounted)
 {
 	bool needScan = false;
 #if JAVA_SPEC_VERSION >= 19
-	jboolean started = J9VMJDKINTERNALVMCONTINUATION_STARTED(vmThread, objectPtr);
-	jboolean finished = J9VMJDKINTERNALVMCONTINUATION_FINISHED(vmThread, objectPtr);
 	J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(vmThread, objectPtr);
-	/**
-	 * We don't scan mounted continuations:
-	 *
-	 * for concurrent GCs, since stack is actively changing. Instead, we scan them during preMount or during root scanning if already mounted at cycle start or during postUnmount (might be indirectly via card cleaning) or during final STW (via root re-scan) if still mounted at cycle end
-	 * for sliding compacts to avoid double slot fixups
-	 *
-	 * For fully STW GCs, there is no harm to scan them, but it's a waste of time since they are scanned during root scanning already.
-	 *
-	 * We don't scan currently scanned for the same collector either - one scan is enough for the same collector, but there could be concurrent scavenger(local collector) and concurrent marking(global collector) overlapping,
-	 * they are irrelevant and both are concurrent, we handle them independently and separately, they are not blocked or ignored each other.
-	 *
-	 * we don't scan the continuation object before started and after finished - java stack does not exist.
-	 */
-	if (started && !finished) {
-		Assert_MM_true(NULL != continuation);
-		needScan = !VM_VMHelpers::isContinuationMountedOrConcurrentlyScanned(continuation, isGlobalGC);
+	if (NULL != continuation)
+	{
+		/**
+		 * We don't scan mounted continuations:
+		 *
+		 * for concurrent GCs, since stack is actively changing. Instead, we scan them during preMount or during root scanning if already mounted at cycle start or during postUnmount (might be indirectly via card cleaning) or during final STW (via root re-scan) if still mounted at cycle end
+		 * for sliding compacts to avoid double slot fixups
+		 * If continuation is currently being mounted by this thread, we must be in preMount/postUnmount callback and must scan.
+		 *
+		 * For fully STW GCs, there is no harm to scan them, but it's a waste of time since they are scanned during root scanning already.
+		 *
+		 * We don't scan currently scanned for the same collector either - one scan is enough for the same collector, but there could be concurrent scavenger(local collector) and concurrent marking(global collector) overlapping,
+		 * they are irrelevant and both are concurrent, we handle them independently and separately, they are not blocked or ignored each other.
+		 *
+		 * we don't scan the continuation object before started and after finished - java stack does not exist.
+		 */
```

Comment:
Would you please reformat this comment to avoid extra long lines? Lines longer than 100 chars are not recommended by coding standards.